lib/orm.php

```diff
@@ -2421,7 +2450,13 @@ protected function build_insert_many( $models, $dirty_column_names ) {
 	public function delete() {
 		$query = [ 'DELETE FROM', $this->quote_identifier( $this->table_name ), $this->add_id_column_conditions() ];
 
-		return self::execute( \implode( ' ', $query ), \is_array( $this->id( true ) ) ? \array_values( $this->id( true ) ) : [ $this->id( true ) ] );
+		// Get the ID.
+		$id = $this->id( true );
+
+		// Flush caches.
+		self::flush_group_caches( $this->table_name );
+
+		return self::execute( \implode( ' ', $query ), \is_array( $id ) ? \array_values( $id ) : [ $id ], false );
```

Comment:
Setting the `$id` var here is not required for this PR. The only relevant line here is the one that flushes the cache.
However, it seemed reasonable to use a var instead of calling `$this->id( true )` 3 times. Just a micro-optimisation 👍